src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -616,7 +619,7 @@ private void ProduceStart(bool appCompleted)
 
         protected Task TryProduceInvalidRequestResponse()
         {
-            if (_requestProcessingStatus == RequestProcessingStatus.RequestStarted && _requestRejected)
```

Comment:
Why did you change this? I'm surprised this hasn't caused any tests to fail.

This change could cause two 4XX responses to be written for a single request. Here's the scenario:

1. There's an invalid request body (e.g. BadChunkSuffix)
2. The app swallows the exception from ReadAsync and completes successfully without writing to the response body.
3. ProduceEnd() is called inside the while loop causing a 4XX to be written.
4. Reset() is called inside the while loop, resetting _requestProcessingStatus to RequestPending
5. The while loop exits because SetBadRequestState sets _requestProccessingStopping to true.
6. TryProduceInvalidRequestResponse() is called in the outer finally of RequestProcessingAsync() causing a 4XX to be written *again*.

Can you add a test to verify this isn't regressed and change it back?